Automattic/wp-calypso
# Jetpack Manage: Adding a new sites submenu to the Jetpack Cloud sidebar navigation

Related to https://github.com/Automattic/jetpack-manage/issues/286

## Proposed Changes

* This PR adds a new sites submenu to the Jetpack Cloud sidebar navigation.
* The submenu items go to the following pages:
  * Needs attention => sites?issue_types=all_issues
  * Favorites => /sites/favorites
  * All => /sites
 * The default sub-menu item is now 'all'. This can be changed to 'needs attention' for example, if needed.
* Challenges in implementation included the 'is-active' menu item behaviour - because all sub-menu items are technically part of the same page all would be highlighted with the previous is-active behaviour, but the implementation turned out to be a one-line change in sites.tsx (changing the default isSelected behaviour there).
* This PR does not include an icon with the number of sites needing attention as that can be followed up with later when an API exists to get the information (it would involve a workaround otherwise). See p1707910367775269/1707846443.511099-slack-CTXBP902X for more context on that decision.


Relevant links:
PT: pfunGA-4t-p2

## Testing Instructions

* To test, apply this PR and navigate to `http://jetpack.cloud.localhost:3000/sites`. Alternatively, when the parent PR is merged, you may be able to use the banner to select the new V2 sites dashboard.
* Click on the 'Sites' menu item, and see the sidebar visible. The default sub menu item is currently 'All'. On clicking each menu item, that should then show as the new active menu item.
* The menu items have the behaviour covered under 'Proposed Changes' above.
* Clicking the 'back' chevron should take you to the Overview page.

#### Screenshots

The main sidebar (currently showing Sites V2):

<img width="656" alt="Screenshot 2024-02-15 at 16 17 47" src="https://github.com/Automattic/wp-calypso/assets/16754605/f34126bc-6ab2-48f2-b450-0f16329a92bd">

The sites sidebar:

<img width="485" alt="Screenshot 2024-02-15 at 16 18 33" src="https://github.com/Automattic/wp-calypso/assets/16754605/3d1d9da5-8cd3-454b-ad53-32f38de2d7c7">


## Pre-merge Checklist
